webpack/scenes/ContentViews/Publish/CVPublishFinish.js

```diff
@@ -0,0 +1,167 @@
+  const response = useSelector(state => selectPublishContentViews(state, cvId, versionCount));
+  const status = useSelector(state => selectPublishContentViewStatus(state, cvId, versionCount));
+  const error = useSelector(state => selectPublishContentViewsError(state, cvId, versionCount));
+  const pollResponse = useSelector(state =>
+    selectPublishTaskPoll(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+  const pollResponseStatus = useSelector(state =>
+    selectPublishTaskPollStatus(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+
+  useEffect(() => {
+    setCurrentStep(3);
+    setSaving(true);
+    dispatch(publishContentView({
+      id: cvId,
+      versionCount,
+      description,
+      environment_ids: userCheckedItems.map(item => item.id),
+      is_force_promote: (forcePromote.length > 0),
+    }));
+    setDescription('');
+    setUserCheckedItems([]);
+  }, [dispatch, setSaving, setDescription, setUserCheckedItems, setCurrentStep]);
+
+  useEffect(() => {
+    setSaving(true);
+    const { id } = response;
+    if (id && status === STATUS.RESOLVED) {
+      setSaving(false);
+      dispatch(pollPublishTask(cvVersionPublishKey(cvId, versionCount), response));
+      setPolling(true);
+    } else if (status === STATUS.ERROR) {
+      setSaving(false);
+    }
+  }, [JSON.stringify(response), status, error]);
+
+  const handleEndTask = () => {
+    setCurrentStep(1);
+    setPolling(false);
+    dispatch(handleFinishedTask({
+      cvId,
+      cvPublishVersionKey: cvVersionPublishKey(cvId, versionCount),
+      task: pollResponse,
+      pending: false,
+    }));
+    setIsOpen(false);
+  };
+
+  useEffect(() => {
+    const { state, result } = pollResponse;
+    if (response && (state === 'paused' || result === 'error')) {
+      setTaskErrored(true);
+      setTimeout(() => {
+        handleEndTask();
+      }, 500);
+    }
+    if (response && (state === 'stopped' && result === 'success')) {
+      setTimeout(() => {
+        handleEndTask();
+      }, 500);
+    }
+  }, [JSON.stringify(pollResponse), dispatch, setTaskErrored,
+    setPolling, setIsOpen, pollResponseStatus]);
+
+  if (saving) {
+    return <Loading />;
+  }
+  if (polling && pollResponse) {
+    return (
+      <Grid hasGutter>
+        <GridItem span={12} rowSpan={11}>
+          <Bullseye>
+            <div style={{ height: '500px', width: '500px' }}>
+              <ChartDonut
+                ariaDesc={__('Publishing Content View')}
+                ariaTitle={__('Publishing Content View')}
+                animate={{ duration: 100 }}
+                data={[{ x: 'Completed', y: pollResponse.progress ? pollResponse.progress * 100 : 0 }, { x: 'Pending', y: pollResponse.progress ? (100 - (pollResponse.progress * 100)) : 100 }]}
+                labels={({ datum }) => `${datum.x}: ${datum.y}%`}
+                title={pollResponse.progress ? `${(pollResponse.progress * 100).toFixed(0)} %` : 'Starting..'}
```

Comment:
I was able to reproduce the 00000 sort of issue. For some reason sometimes the API returns a very long decimal value in the task progress result instead of the usual decimal with precision 2 or 3. That was causing this title to be a very long number. .toFixed(0) should fix that since we don't really need decimal precision in % complete.

Reply:
Oooooooh!  Nice :)